### PR TITLE
BUG: Resampling to "B" frequency with closed="right" and label="right" adds empty bins

### LINKED ISSUE #59495: BUG: Resampling to `"B"` frequency with `closed="right"` and `label="right"` adds empty bins
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd

idx = pd.date_range("5Aug2024", "13Aug2024", freq="D")  # Mon to Tue the week after
x = pd.Series(range(len(idx)), idx) + 1

# this gives unexpected output
x.resample("B", closed="right", label="right").sum()

# my expected output of the above is what the following code instead gives
x.resample("B", closed="right", label="right").sum().iloc[:-1]

# curiously, in the following different scenario the output is what I'd expect
x.resample("B", closed="left", label="left").sum()
```

### Issue Description

When resampleing `D` frequency data to `B` frequency using `closed="right"` and `label="right"` I wouldn't expect to get the trailing empty bin. Indeed, resampling similarly with `closed="left"` and `label="left"` doesn't give any leading empty bin (which, even if undesirable, at least would be consistent with the `"right"` behaviour).

### Expected Behavior

The output of the first resampling call in the MRE above should be what is currently output by the second resampling call in the MRE.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit                : d9cdd2ee5a58015ef6f4d15c7226110c9aab8140
python                : 3.11.9.final.0
python-bits           : 64
OS                    : Linux
OS-release            : 5.15.0-105-generic
Version               : #115~20.04.1-Ubuntu SMP Mon Apr 15 17:33:04 UTC 2024
machine               : x86_64
processor             : x86_64
byteorder             : little
LC_ALL                : None
LANG                  : en_US.UTF-8
LOCALE                : en_US.UTF-8

pandas                : 2.2.2
numpy                 : 1.26.4
pytz                  : 2024.1
dateutil              : 2.9.0
setuptools            : 69.5.1
pip                   : 24.0
Cython                : None
pytest                : 8.2.0
hypothesis            : None
sphinx                : None
blosc                 : None
feather               : None
xlsxwriter            : None
lxml.etree            : 5.2.1
html5lib              : None
pymysql               : None
psycopg2              : 2.9.9
jinja2                : 3.1.3
IPython               : 8.22.2
pandas_datareader     : 0.10.0
adbc-driver-postgresql: None
adbc-driver-sqlite    : None
bs4                   : 4.12.3
bottleneck            : None
dataframe-api-compat  : None
fastparquet           : None
fsspec                : None
gcsfs                 : None
matplotlib            : 3.8.4
numba                 : 0.59.1
numexpr               : None
odfpy                 : None
openpyxl              : 3.1.2
pandas_gbq            : None
pyarrow               : 15.0.2
pyreadstat            : None
python-calamine       : None
pyxlsb                : None
s3fs                  : None
scipy                 : 1.13.0
sqlalchemy            : 2.0.29
tables                : None
tabulate              : 0.9.0
xarray                : None
xlrd                  : None
zstandard             : None
tzdata                : 2024.1
qtpy                  : 2.4.1
pyqt5                 : None

</details>

### PR DESCRIPTION
**Pull request by natmokval**
- [x] closes #59495
- [x] Tests added and passed
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/v3.0.0.rst`

fixed bug: Resampling to frequency='B' with closed='right'a nd label='right'

### CODE REVIEW

**Comment by natmokval** [comment]
@MarcoGorelli, seems like the reproducible example was affected by the changes in your PR #55283. I corrected the definition of `_get_time_bins` to get the expected behavior in the example. However, this caused some of your tests to fail, so I updated the expected output. Would you mind reviewing my changes?

**Comment by MarcoGorelli** [comment]
thanks for the ping, will take a look next week

File: pandas/core/resample.py [hunk 1]
```diff
@@ -2173,7 +2173,11 @@ def _get_time_bins(self, ax: DatetimeIndex):
         if self.closed == "right":
             labels = binner
             if self.label == "right":
-                labels = labels[1:]
+                if binner.freqstr == "B":
+                    labels = labels[1:-1]
+                    bins = bins[0:-1]
+                else:
+                    labels = labels[1:]
```
Comment (MarcoGorelli) [inline pandas/core/resample.py:2180]: this isn't so clear to me - could you explain what happens here and how it fixes the issue please?

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**